subprojects/signing/src/main/java/org/gradle/plugins/signing/SigningExtension.java

```diff
@@ -412,7 +414,14 @@ public void execute(Publication publication) {
     }
 
     private <T extends PublicationArtifact> Sign createSignTaskFor(final PublicationInternal<T> publicationToSign) {
-        final Sign signTask = project.getTasks().create(determineSignTaskNameForPublication(publicationToSign), Sign.class, new Action<Sign>() {
+        final String signTaskName = determineSignTaskNameForPublication(publicationToSign);
+
+        final Task existingSignTask = project.getTasks().findByName(signTaskName);
```

Comment:
> You cannot call `findByName` as it will make task creation eager for _all_ tasks. See explanation [here](https://docs.gradle.org/current/userguide/task_configuration_avoidance.html#sec:old_vs_new_configuration_api_overview).

Ok then the approach would be to use a `try-catch` block since both `create` and `named` are throwing and exception.

> Also, I wonder if having a check just on the name is not too little. While we do not want to fail for multiple invocations, we still need to fail if there is a name conflict but different tasks under that common name.

What check would you add? It is possible to have multiple `Sign` tasks, e.g. one per publication, so adding a check by type would not be possible.

> So maybe the signing extension should instead remember what it was asked to sign and skip on a duplicate request.

I believe the issue happens when registering a task with a duplicate name, not when the sign action is executed. Are we missing any case?